src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -460,6 +460,57 @@ class HttpGoogleServicesDAO(
     }
   }
 
+  def getBilling(credential: Credential) = {
+    new Cloudbilling.Builder(httpTransport, jsonFactory, credential).build()
+  }
+
+  /**
+    * NOTE: This function will returns "false" in both of the following cases:
+    * if you don't have sufficient scopes
+    *   - Google's JSON response body will contain "message": "Request had insufficient authentication scopes."
+    * if you're not authorized to see the billing account
+    *   - Google's JSON response body will contain "message" : "The caller does not have permission"
+    */
+  protected def credentialOwnsBillingAccount(credential: Credential, billingAccountName: String): Future[Boolean] = {
+    val fetcher = getBilling(credential).billingAccounts().get(billingAccountName)
+    retryWithRecoverWhen500orGoogleError(() => {
+      blocking {
+        executeGoogleRequest(fetcher)
+      }
+      true //if the request succeeds, it has access.
+    }) {
+      case e: HttpResponseException if e.getStatusCode == StatusCodes.Forbidden.intValue => false
+    }
+  }
+
+  protected def listBillingAccounts(credential: Credential): Future[Seq[BillingAccount]] = {
+    val fetcher = getBilling(credential).billingAccounts().list()
+    retryWithRecoverWhen500orGoogleError(() => {
+      val list = blocking {
+        executeGoogleRequest(fetcher)
+      }
+      list.getBillingAccounts.toSeq
+    }) {
+      case e: HttpResponseException if e.getStatusCode == StatusCodes.Forbidden.intValue =>
+        throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.Forbidden, e.getContent))
+    }
+  }
+
+  def listBillingAccounts(userInfo: UserInfo): Future[Seq[RawlsBillingAccount]] = {
+    val cred = getUserCredential(userInfo)
+    val billingSvcCred = getBillingServiceAccountCredential
+    listBillingAccounts(cred) flatMap { accountList =>
+      Future.sequence(accountList map { acct =>
```

Comment:
for N accounts owned by the user, this makes N+1 requests to Google. Could you, instead, query for 1) the list of accounts for the user, 2) the list of accounts for the FC billing account, and compare the two lists?

My approach may be LESS performant if FC owns a ton of accounts and users typically have very few. Do we have a sense of how many billing accounts users will have? Probably not a lot ...